lib/clients/ask-smapi-client/index.js

```diff
@@ -0,0 +1,83 @@
+const { CustomSmapiClientBuilder } = require('ask-smapi-sdk');
+
+const AppConfig = require('@src/model/app-config');
+const AuthorizationController = require('@src/controllers/authorization-controller');
+const DynamicConfig = require('@src/utils/dynamic-config');
+const Messenger = require('@src/view/messenger');
+const jsonView = require('@src/view/json-view');
+
+const _stringifyHeaders = (item) => {
+    let headers = {};
+    item.headers.forEach(h => {
+        headers[h.key] = h.value;
+    });
+    headers = JSON.stringify(headers);
+    return { ...item, headers };
+};
+
+/**
+ * @param {Object}request Request object
+ * @param {Array<{key, value}>} request.headers Array of headers
+ * @param {string} request.method HTTP method
+ * @param {string} request.url HTTP url
+ * @param {Object} request.body HTTP body
+ */
+const logRequest = (req) => {
+    Messenger.getInstance().info('REQUEST:');
+    Messenger.getInstance().info(jsonView.toString(_stringifyHeaders(req)));
```

Comment:
just a head up that this Messenger.info will break the experiences when it's not a pass-through smapi command. Messenger also has debug method, which should be the better one to use here - logs will be flushed at the end of Messenger instance.

Reply:
try switching it to Messenger debug - no output